generator_process/intents/prompt_to_image.py

```diff
@@ -196,4 +196,76 @@ def update(self, n=1):
+
+    def image_writer(image, seed, upscaled=False, first_seed=None):
+        # Only use the non-upscaled texture, as upscaling is a separate step in this addon.
+        if not upscaled:
+            self.send_action(Action.IMAGE, shared_memory_name=self.share_image_memory(image), seed=seed, width=image.width, height=image.height)
+
+    while True:
+        self.check_stop()
+
+        self.send_info("Generating...")
+        
+        seed = random.randrange(0, 4294967295) if args['seed'] is None else args['seed']
+        def realtime_viewport_init_image():
+            return None # Realtime viewport is not currently available.
+            if args['init_img_shared_memory'] is not None:
+                init_img_memory = SharedMemory(args['init_img_shared_memory'])
+                shared_init_img = Image.frombytes('RGBA', (args['init_img_shared_memory_width'], args['init_img_shared_memory_height']), init_img_memory.buf.tobytes())
+                shared_init_img = shared_init_img.resize((512, round(((shared_init_img.height / shared_init_img.width) * 512) / 64)*64))
+                init_img_memory.close()
+                return shared_init_img
+            return None
+        shared_init_img = realtime_viewport_init_image()
+        answers = stability_inference.generate(
+            prompt=args['prompt'],
+            init_image=shared_init_img if shared_init_img is not None else (Image.open(args['init_img']) if args['init_img'] is not None else None),
+            # mask_image: Optional[Image.Image] = None,
+            width=shared_init_img.width if shared_init_img is not None else args['width'],
+            height=shared_init_img.height if shared_init_img is not None else args['height'],
+            start_schedule=1.0 * args['strength'],
+            end_schedule=0.01,
+            cfg_scale=args['cfg_scale'],
+            sampler=algorithms[args['sampler_name']],
+            steps=args['steps'],
+            seed=seed,
+            samples=args['iterations'],
+            # safety: bool = True,
+            # classifiers: Optional[generation.ClassifierParameters] = None,
+            # guidance_preset: generation.GuidancePreset = generation.GUIDANCE_PRESET_NONE,
+            # guidance_cuts: int = 0,
+            # guidance_strength: Optional[float] = None,
+            # guidance_prompt: Union[str, generation.Prompt] = None,
+            # guidance_models: List[str] = None,
+        )
+
+        for answer in answers:
+            for artifact in answer.artifacts:
+                if artifact.finish_reason == interfaces.gooseai.generation.generation_pb2.FILTER:
+                    self.send_exception(False, "Your request activated DreamStudio's safety filter. Please modify your prompt and try again.")
+                if artifact.type == interfaces.gooseai.generation.generation_pb2.ARTIFACT_IMAGE:
+                    response = Image.open(io.BytesIO(artifact.binary))
+                    image_writer(response, seed)
```

Comment:
Iterations returns same seed for all images.
```suggestion
                    image_writer(response, artifact.seed)
```